change endpoint

adesamp opened the pull request:
change endpoint with news api

adeds committed:
change endpoint

mochadwi commented on mobile-ui/src/main/java/org/buffer/android/boilerplate/ui/model/ArticleViewModel.kt:
```diff
@@ -0,0 +1,12 @@
+package org.buffer.android.boilerplate.ui.model
+
+
+class ArticleViewModel(
```
Looks like I'm forgot to tell you about this.

`*ViewModel` seems conflicted with `ViewModel` from Architecture Components

mochadwi replied:
How about `*Intent`? @adesamp

mochadwi commented on remote/src/main/java/org/buffer/android/boilerplate/remote/BufferooServiceFactory.kt:
```diff
@@ -24,7 +24,7 @@ object BufferooServiceFactory {
     private fun makeBufferooService(okHttpClient: OkHttpClient, gson: Gson): BufferooService {
         val retrofit = Retrofit.Builder()
-                .baseUrl("https://joe-birch-dsdb.squarespace.com/s/")
+                .baseUrl("https://newsapi.org/v2/")
```
It is recommended to store the `BASE_URL` inside global `gradle.properties` also @adesamp

mochadwi commented on remote/src/main/java/org/buffer/android/boilerplate/remote/model/ArticleModel.kt:
```diff
@@ -0,0 +1,32 @@
+/**
+ * Representation for a [ArticleModel] fetched from the API
+ */
+class ArticleModel(
```
For remote (or data) module. The naming would be like this:
1. domain -> Article
2. data/remote/cache -> ArticleEntity (db), ArticleResponse/ArticleRequest/ArticleParam (network/json)
3. ui/presentation -> ArticleIntent

Why I choose using `ArticleIntent` instead of `ArticleView` because we might creating a custom view, thus will be conflicted with POJO/data class.

cc @adesamp

mochadwi commented on remote/src/main/java/org/buffer/android/boilerplate/remote/ArticleService.kt:
```diff
@@ -0,0 +1,22 @@
+    @GET("top-headlines?country=id")
+    fun getArticles(@Query("apiKey")key: String): Flowable<EverythingResponse>
+
+    class EverythingResponse(
```
did you mean `ArticleResponse`?

mochadwi commented on presentation/src/main/java/org/buffer/android/boilerplate/presentation/browse/model/ArticleView.kt:
```diff
@@ -0,0 +1,12 @@
+package org.buffer.android.boilerplate.presentation.browse.model
+
+
+class ArticleView(
```
Well this `presentation` module might be good to be combined with `mobile-ui` that is assigned to me haha

mochadwi commented on presentation/src/main/java/org/buffer/android/boilerplate/presentation/browse/BrowseProcessor.kt:
```diff
@@ -2,12 +2,14 @@ package org.buffer.android.boilerplate.presentation.browse
+                                          private val getArticles: GetArticles) {
 
-    private val conversationsProcessor: ObservableTransformer<
+    private val conversationsProcessorOld: ObservableTransformer<
```
Did you use this? If not, then it is safe to delete

mochadwi commented on mobile-ui/src/main/java/org/buffer/android/boilerplate/ui/browse/BrowseAdapter.kt:
```diff
@@ -9,20 +10,19 @@ import android.widget.TextView
+        val article = articles[position]
+        holder.nameText.text = article.author
+        holder.titleText.text = article.title
         Glide.with(holder.itemView.context)
```
This sample of using glide might also needs to be optimized, see here for details: https://stackoverflow.com/a/32887693/3763032

mochadwi commented on cache/src/main/java/org/buffer/android/boilerplate/cache/db/constants/ArticleConstants.kt:
```diff
@@ -0,0 +1,14 @@
+/**
+ * Defines constants for the Article Table
+ */
+object ArticleConstants {
```
Wait, is there a reason why using `object` constant here? @adesamp 

See here about constant: 

https://sampingan.atlassian.net/wiki/spaces/S/pages/38141970/Android+Code+Convention#Writing-meaningful-%E2%80%9CConstant%E2%80%9D

mochadwi reviewed (changes requested):
There's a few notes to be changed

adeds committed:
changes by request

adeds committed:
remove unused task

mochadwi commented on cache/src/main/java/org/buffer/android/boilerplate/cache/dao/CachedArticleDao.kt:
```diff
@@ -4,19 +4,22 @@ import android.arch.persistence.room.Dao
-import org.buffer.android.boilerplate.cache.model.CachedArticle
+import org.buffer.android.boilerplate.cache.model.ArticleEntity
+
+const val TABLE_NAME = "article"
+const val QUERY_ARTICLES = "SELECT * FROM $TABLE_NAME"
+const val DELETE_ALL_ARTICLES = "DELETE FROM $TABLE_NAME"
```
nice

mochadwi commented on cache/src/main/java/org/buffer/android/boilerplate/cache/mapper/ArticleEntityMapper.kt:
```diff
@@ -1,18 +1,17 @@
  */
 open class ArticleEntityMapper @Inject constructor() :
-        EntityMapper<CachedArticle, ArticleEntity> {
+        EntityMapper<org.buffer.android.boilerplate.cache.model.ArticleEntity, ArticleEntity> {
```
I forgot there's already a name with `ArticleEntity` ya di data? My bad @adesamp

mochadwi commented on cache/src/main/java/org/buffer/android/boilerplate/cache/model/ArticleEntity.kt:
```diff
@@ -2,14 +2,13 @@ package org.buffer.android.boilerplate.cache.model
-@Entity(tableName = ArticleConstants.TABLE_NAME)
-data class CachedArticle(
+@Entity(tableName = TABLE_NAME)
+data class ArticleEntity(
```
It is safe to use `CachedArticle` I guess, but for in the future, this module will be merged to `data` module

mochadwi commented on mobile-ui/src/main/java/org/buffer/android/boilerplate/ui/browse/BrowseActivity.kt:
```diff
@@ -84,6 +84,7 @@ class BrowseActivity : AppCompatActivity(),
     }
 
     private fun setupBrowseRecycler() {
+        browseAdapter = BrowseAdapter(Glide.with(this))
```
nice

mochadwi commented on mobile-ui/src/main/java/org/buffer/android/boilerplate/ui/mapper/ArticleMapper.kt:
```diff
@@ -1,19 +1,17 @@
-    override fun mapToViewModel(article: ArticleView): ArticleViewModel {
-        return ArticleViewModel(
+    override fun mapToViewModel(article: ArticleView): ArticleIntent {
+        return ArticleIntent(
```
nice

mochadwi commented on remote/src/main/java/org/buffer/android/boilerplate/remote/ArticleRemoteImpl.kt:
```diff
@@ -15,14 +15,16 @@ import javax.inject.Inject
 
-    override fun getArticles(): Flowable<List<ArticleEntity>> {
+    init {
         val klass = Class.forName("org.buffer.android.boilerplate.ui.BuildConfig")
```
I see, awesome implementation, using reflection to detect the runtime package inside `mobile-ui`. But, I'll try to find an alternative, this is tends to be painful if there's a package name changed

mochadwi replied:
except this is used for `andorid dynamice feature`

mochadwi replied:
or using NDK instead to call native c++ method in java/kotlin, see here #6

mochadwi reviewed (commented):
Thanks @adesamp great works

mochadwi committed:
refactor(data): Remote & Cache reverted

- Both remote & cache module will be merged into data instead of renaming the
model for cache, remote & data
